core/range-controller.js

```diff
@@ -773,7 +773,13 @@ var RangeController = exports.RangeController = Montage.specialize( /** @lends R
                 // remove all values from the selection that were removed (but
                 // not added back)
                 minus.deleteEach(plus, equals);
-                if (this.selection) {
+
+                if (this.selection.length) {
+                    // ensure selection always has content
+                    if (this.content.length && this.avoidsEmptySelection && !this.multiSelect) {
```

Comment:
Ok I got it. You assume that whether the selection is not empty technically the content should not be empty as well. Looks like we have some security for the selection property: https://github.com/rayshan/montage/blob/pr-rangecontroller/core/range-controller.js#L468